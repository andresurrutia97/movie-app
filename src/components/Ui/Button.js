import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ children, onClick, type }) => {
  const baseClass = 'w-full focus:outline-none text-white py-2 px-4 rounded-md';

  let classes;

  switch (type) {
    case 'outlined':
      classes = [baseClass, 'border border-2 custom-blue-border bg-black'].join(
        ' '
      );
      break;

    default:
      classes = [baseClass, 'custom-blue-bg'].join(' ');
  }

  return (
    <button type="button" onClick={onClick} className={classes}>
      {children}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.string,
  onClick: PropTypes.func,
  type: PropTypes.string,
};

Button.defaultProps = {
  children: 'Click me',
  onClick: () => {},
  type: '',
};

export default Button;
