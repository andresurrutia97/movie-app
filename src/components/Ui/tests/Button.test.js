import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Button from '../Button';

describe('<Button/>', () => {
  const mockfn = jest.fn();
  const component = <Button onClick={mockfn}>Test</Button>;

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(component);
  });

  it('should render correctly ', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call the mockFn function when click', () => {
    wrapper.simulate('click');
    expect(mockfn).toBeCalledTimes(1);
  });

  it('should have "test" as text', () => {
    expect(wrapper.props().children).toEqual('Test');
  });
});
