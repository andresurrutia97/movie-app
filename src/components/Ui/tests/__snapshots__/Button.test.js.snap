// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should render correctly  1`] = `
<button
  className="w-full focus:outline-none text-white py-2 px-4 rounded-md custom-blue-bg"
  onClick={[MockFunction]}
  type="button"
>
  Test
</button>
`;
